pkg/apiserver/endpoints/filters/path_rewriter_test.go

```diff
@@ -0,0 +1,45 @@
+package filters
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_WithPathRewriters(t *testing.T) {
+	mockHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(r.URL.Path))
+	})
+
+	rewriters := []PathRewriter{
+		{
+			Pattern: regexp.MustCompile(`(/apis/scope.grafana.app/.*/query)(/.*)`),
+			ReplaceFunc: func(matches []string) string {
+				return matches[1]
+			},
+		},
+	}
+	handler := WithPathRewriters(mockHandler, rewriters)
+
+	t.Run("should rewrite path", func(t *testing.T) {
+		req, err := http.NewRequest("GET", "/apis/scope.grafana.app/namespaces/stack-1234/query/blah", nil)
```

Comment:
Are we worried about conflicts for `/query` and other kinds?

Reply:
IIUC, it will depend on the regexp -- the one we would change is pretty locked down:
https://github.com/grafana/grafana/pull/87457/files#diff-36b432133a11278c45036e382509c13ef488087cd22e69f8737348100443af0bR32

Reply:
yep, this is just an example in the tests